pkg/providers/tinkerbell/stack/stack.go

```diff
@@ -273,6 +288,17 @@ func (s *Installer) getBootsEnv(bundle releasev1alpha1.TinkerbellStackBundle, ti
 	if s.registryMirror != nil {
 		localRegistry := net.JoinHostPort(s.registryMirror.Endpoint, s.registryMirror.Port)
 		extraKernelArgs = fmt.Sprintf("%s insecure_registries=%s", extraKernelArgs, localRegistry)
+		if s.registryMirror.Authenticate {
+			username, password, _ := config.ReadCredentials()
+			return map[string]string{
+				"DATA_MODEL_VERSION":        "kubernetes",
+				"TINKERBELL_TLS":            "false",
+				"TINKERBELL_GRPC_AUTHORITY": fmt.Sprintf("%s:%s", tinkServerIP, grpcPort),
+				"BOOTS_EXTRA_KERNEL_ARGS":   extraKernelArgs,
+				"REGISTRY_USERNAME":         username,
+				"REGISTRY_PASSWORD":         password,
+			}
+		}
```

Comment:
Instead of doing multiple returns, can you declare a map outside the if and just append the registry user/pass env to that?